### PR TITLE
Add Huggingface integration documentation

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add Huggingface integration documentation.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: docs/source/overview/options.md [hunk 1]
```diff
@@ -144,6 +144,9 @@ find more details in [Olive Models](https://microsoft.github.io/Olive/api/models
             ```
             For cases where you do not want to use the huggingface model but want to use the huggingface dataset, you can provide `dataset` config only like above.
 
+        - `model_loading_args: [dict]`: Arguments to pass to the `from_pretrained` method of the model class. Refer to [this documentation](https://huggingface.co/docs/transformers/main_classes/model#transformers.PreTrainedModel.from_pretrained).
```
Comment (devang-ml) [inline docs/source/overview/options.md:147]: Folks are familiar with HF's from_pretrained() API, so rename it to be 'hf_from_pretrained_args' for clarity.

File: olive/model/hf_utils.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ class HFComponent(ConfigBase):
 class HFModelLoadingArgs(ConfigWithExtraArgs):
     """Arguments to pass to the `from_pretrained` method of the model class.
```
Comment (devang-ml) [inline olive/model/hf_utils.py:32]: Same way rename this to HFFromPretrainedArgs

File: docs/source/features/huggingface_model_optimization.md [hunk 3]
```diff
@@ -1,58 +1,98 @@
...
+        }
+    }
 }
 ```
-Please refer to [metrics](../overview/options.md#metrics) for more details.
 
-### Custom components config
+#### Optimum model
+Optimum model is a special case of PyTorhc model. By specifying `OptimumModel` as `type`, the model path would be the model name. And add model components name. Olive will get the components from Huggingface hub:
+
```
Comment (devang-ml) [inline docs/source/features/huggingface_model_optimization.md:55]: typo 'PyTorhc'

File: docs/source/overview/options.md (hunk 1 above)
Comment (xiaoyu-work) [reply docs/source/overview/options.md:147]: Sure, will send a separate PR for this change.

File: docs/source/features/huggingface_model_optimization.md (hunk 3 above)
Comment (xiaoyu-work) [reply docs/source/features/huggingface_model_optimization.md:55]: Fixed.

File: docs/source/overview/options.md (hunk 1 above)
Comment (xiaoyu-work) [reply docs/source/overview/options.md]: PR link: https://github.com/microsoft/Olive/pull/785

File: docs/source/features/huggingface_model_optimization.md [hunk 4]
```diff
@@ -1,58 +1,98 @@
...
-    - Load model in which user can just provide the model name and task name. Olive will automatically load the model from Huggingface model hub for specific task.
-    - Load specific tokenizer and data processor. Currently, we simplify the user experience only for `text-classification` task, which will be demonstrated in the following examples. *For other tasks, it is under development actively.*
-3. The task specific dataset: is hosted in [Huggingface datasets](https://huggingface.co/datasets), e.g. `glue` dataset for text classification task.
-4. Metric name: is supported by [Huggingface evaluate](https://huggingface.co/docs/evaluate/index). User can refer to [huggingface metrics](https://huggingface.co/metrics) for full metrics list.
-
-## Example
-
-### Model config and dataset config
-Take `bert-base-uncased` as an example, user can specify task name as `text-classification` and dataset as `glue` to form the Huggingface config `hf_config` as follows:
+This documents outlines the integrations between Olive and Huggingface. Discover how to use Huggingface resources within Olive.
```
Comment (trajepl) [inline docs/source/features/huggingface_model_optimization.md:4]: `documents` -> `document`

File: docs/source/features/huggingface_model_optimization.md [hunk 5]
```diff
@@ -1,58 +1,98 @@
...
+            "task": "text-classification"
+        }
+    }
 }
 ```
-Please refer to [metrics](../overview/options.md#metrics) for more details.
 
-### Custom components config
+#### Optimum model
+Optimum model is a special case of PyTorch model. By specifying `OptimumModel` as `type`, the model path would be the model name. And add model components name. Olive will get the components from Huggingface hub:
```
Comment (trajepl) [inline docs/source/features/huggingface_model_optimization.md:54]: `. And add`  -> `and`?

Comment (xiaoyu-work) [reply docs/source/features/huggingface_model_optimization.md]: Updated to make it clear

File: docs/source/features/huggingface_model_optimization.md (hunk 4 above)
Comment (xiaoyu-work) [reply docs/source/features/huggingface_model_optimization.md]: Fixed.

File: docs/source/features/huggingface_model_optimization.md [hunk 6]
```diff
@@ -1,58 +1,98 @@
...
+
+## hf_config
+If you want to optimize a Huggingface model, or evaluate a Huggingface model, you will need `hf_config` defined in your `input_model` section. Please refer to [this section](../overview/options.md#input-model-information) for detailed parameters of `hf_config`.
+
+Here is how you can use `hf_config`:
+
+### Model config loading
+Olive can automatically retrieve model configurations from Huggingface hub:
+
+- Olive retrieves model [configuration](https://huggingface.co/docs/transformers/main/en/model_doc/auto#transformers.AutoConfig) from transfromers for future usage.
```
Comment (jambayk) [inline docs/source/features/huggingface_model_optimization.md:14]: typo: transformers

Comment (xiaoyu-work) [reply docs/source/features/huggingface_model_optimization.md]: Fixed.

**Review by jambayk — DISMISSED** [review]

**Pull request merged**